src/FixtureDownloader.ts

```diff
@@ -0,0 +1,16 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+import { Downloader } from './Downloader';
+
+const FIXTURE_DIR = path.resolve(__dirname, '../test/fixtures');
+
+export class FixtureDownloader implements Downloader<any> {
+  async download(url: string, targetFilePath: string) {
+    await fs.ensureDir(path.dirname(targetFilePath));
+    const fixtureFile = path.join(FIXTURE_DIR, path.basename(targetFilePath));
+    if (await fs.pathExists(fixtureFile)) {
+      await fs.copy(fixtureFile, targetFilePath);
+    }
```

Comment:
Should there be an `else` clause which throws an error? Seems like a potential divergence from `GotDownloader`, where one will throw an error on a bad URL and the other won't.

Reply:
It could, but none of the tests use it currently, and it probably shouldn't. The lack of a fixture isn't something that really needs to be tested, because at that point you're testing the scaffolding.

The `GotDownloader` spec tests for nonexistent URLs, which seems like a better way to do it.

Reply:
> It could, but none of the tests use it currently, and it probably shouldn't.

I disagree. It's pretty foreseeable that someone may want to write a test that verifies how `downloadArtifact` handles a bad URL. I was tempted to add one while trying to track down an uncaught error in promise from the Electron npm install script, to check that the error wasn't from within `downloadArtifact` itself.

If `FixtureDownloader` doesn't throw in that case, it will lead to unexplained behavior if someone does try to write that test, as they'll get an unexpected result, and will have to spend time tracking it down. Since the swap to `FixtureDownloader` is in `src/` rather than `test/`, it's not immediately clear that a different downloader is being used.

I'd suggest at least a warning be printed in `else`, that way if someone creates a test which would hit a missing fixture, they are at least warned as to what is happening.

> The lack of a fixture isn't something that really needs to be tested, because at that point you're testing the scaffolding.

What would be being tested is how `downloadArtifact` handles a bad URL, not the lack of a fixture.